application/shoot.h

```diff
@@ -61,9 +61,16 @@
 #define FRICTION_MOTOR_SPEED  25.0f
 #endif
 
-#define SEMI_AUTO_FIRE_TRIGGER_SPEED 10.0f
-#define AUTO_FIRE_TRIGGER_SPEED      15.0f
-#define READY_TRIGGER_SPEED         5.0f
+// max speed of M3508 is 26.99m/s for one motor, 26.2m/s for one motor during test
+#if FRICTION_MOTOR_SAFETY_GUARD
+#define FRICTION_MOTOR_SPEED  1.0f
+#else
+#define FRICTION_MOTOR_SPEED  25.0f
+#endif
+
+#define SEMI_AUTO_FIRE_TRIGGER_SPEED (15.0f * TRIGGER_MOTOR_TO_WHEEL_GEAR_RATIO)
+#define AUTO_FIRE_TRIGGER_SPEED      (15.0f * TRIGGER_MOTOR_TO_WHEEL_GEAR_RATIO)
+#define READY_TRIGGER_SPEED         (5.0f * TRIGGER_MOTOR_TO_WHEEL_GEAR_RATIO)
```

Comment:
The old code was right, `TRIGGER_MOTOR_TO_WHEEL_GEAR_RATIO` is already compensated inside `TRIGGER_MOTOR_ECD_TO_ANGLE`. Also, these macros define the target speed in rpm, so they should not be multiplied by the gear ratio after all.